src/lib/util.ts

```diff
@@ -18,84 +27,59 @@ export function firstMatchingToken<T>(tokenList: string, validValues: T[]): T |
   return parseTokenList(tokenList).find((value) => validValues.includes(value as T)) as T | undefined;
 }
 
-/** Are we able to traverse the DOM? */
-export function isHTMLElement(element: HTMLElement | VirtualElement): boolean {
-  return typeof HTMLElement !== 'undefined' && element instanceof HTMLElement;
-}
-
-/** Normalize HTML Elements */
-export function virtualizeElement(element: HTMLElement | VirtualElement): VirtualElement {
-  // handle HTMLElement or VirtualElement
-  let tagName = '' as TagName;
-  let attributes: VirtualElement['attributes'];
-
-  if (isHTMLElement(element)) {
-    tagName = element.tagName.toLowerCase() as TagName;
-    attributes = {};
-    for (let i = 0; i < (element as HTMLElement).attributes.length; i++) {
-      // biome-ignore lint/style/noNonNullAssertion: This is guaranteed
-      const { name } = (element as HTMLElement).attributes[i]!;
-      attributes[name] = (element as HTMLElement).getAttribute(name);
-    }
-    return { tagName, attributes };
-  }
-
-  if (!element || typeof element !== 'object' || Array.isArray(element) || typeof element.tagName !== 'string') {
-    throw new Error(`Expected { tagName, [attributes] } object, received ${JSON.stringify(element)}`);
+export function getTagName(element: Element | VirtualElement): TagName {
+  if (typeof Element !== 'undefined' && element instanceof Element) {
```

Comment:
I ended up with this line of code a lot, to split the code path to run in the DOM. Even though this would be a good helper, TypeScript gets lost and you end up having to make the same type assertions manually, more or less. There’s probably a cleaner way to separate DOM code from non- but it didn’t come to me while writing this.